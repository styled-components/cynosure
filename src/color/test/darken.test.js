// @flow
import darken from '../darken'

describe('darken', () => {
  it('should darken a color by 20%', () => {
    expect(darken(0.2, '#444')).toMatchSnapshot()
  })

  it('should darken an 8-digit hex color by 20%', () => {
    expect(darken(0.2, '#6564CDB3')).toMatchSnapshot()
  })

  it('should darken a color with opacity by 20%', () => {
    expect(darken(0.2, 'rgba(101,100,205,0.7)')).toMatchSnapshot()
  })

  it('should darken a color with a value of 255 and opacity by 10%', () => {
    expect(darken(0.1, 'rgba(255,140,140,0.7)')).toMatchSnapshot()
  })

  it('should darken a color but not go below 0', () => {
    expect(darken(0.8, 'rgba(40,20,10,0.7)')).toMatchSnapshot()
  })

  it('should darken a color by when passed a string for amount', () => {
    expect(darken('0.2', '#444')).toMatchSnapshot()
  })
})
