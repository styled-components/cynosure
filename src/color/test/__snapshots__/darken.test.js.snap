// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`darken should darken a color but not go below 0 1`] = `"rgba(0,0,0,0.7)"`;

exports[`darken should darken a color by 20% 1`] = `"#111"`;

exports[`darken should darken a color by when passed a string for amount 1`] = `"#111"`;

exports[`darken should darken a color with a value of 255 and opacity by 10% 1`] = `"rgba(255,89,89,0.7)"`;

exports[`darken should darken a color with opacity by 20% 1`] = `"rgba(51,50,153,0.7)"`;

exports[`darken should darken an 8-digit hex color by 20% 1`] = `"rgba(51,50,153,0.7)"`;
